app/controllers/sessions_controller.rb

```diff
@@ -9,15 +9,22 @@ def new
   def login
     user = User.find_by(email: params[:email])
     if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
       flash[:notice] = "Logged in as #{user.name}"
-      redirect_to root_path
+      redirect_to '/user/dashboard'
     else
-      flash[:notice] = 'Your email or password was incorrect!'
+      flash[:notice] = "Sorry, we don't recognize those credentials."
       render :new
     end
   end
 
   def logout
+    NpcModel.where(is_saved: 0).ids.map do |id|
+      NpcModelArmor.where(npc_model_id: id).destroy_all
+      NpcModelPack.where(npc_model_id: id).destroy_all
+      NpcModelWeapon.where(npc_model_id: id).destroy_all
+    end
+    NpcModel.where(is_saved: 0).destroy_all
```

Comment:
This is done so that the DB doesn't get overrun with NPC's that have been generated, but that haven't been "saved" by any user.